src/Neo.VM/JumpTable/JumpTable.Splice.cs

```diff
@@ -0,0 +1,106 @@
+// Copyright (C) 2015-2024 The Neo Project.
+//
+// JumpTable.Splice.cs file belongs to the neo project and is free
+// software distributed under the MIT software license, see the
+// accompanying file LICENSE in the main directory of the
+// repository or http://www.opensource.org/licenses/mit-license.php
+// for more details.
+//
+// Redistribution and use in source and binary forms with or without
+// modifications are permitted.
+
+using System;
+using System.Runtime.CompilerServices;
+
+namespace Neo.VM
+{
+    public partial class JumpTable
+    {
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public virtual void NewBuffer(ExecutionEngine engine, Instruction instruction)
+        {
+            int length = (int)engine.Pop().GetInteger();
+            engine.Limits.AssertMaxItemSize(length);
+            engine.Push(new Types.Buffer(length));
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public virtual void Memcpy(ExecutionEngine engine, Instruction instruction)
+        {
+            int count = (int)engine.Pop().GetInteger();
+            if (count < 0)
+                throw new InvalidOperationException($"The value {count} is out of range.");
+            int si = (int)engine.Pop().GetInteger();
+            if (si < 0)
+                throw new InvalidOperationException($"The value {si} is out of range.");
+            ReadOnlySpan<byte> src = engine.Pop().GetSpan();
+            if (checked(si + count) > src.Length)
+                throw new InvalidOperationException($"The value {count} is out of range.");
+            int di = (int)engine.Pop().GetInteger();
+            if (di < 0)
+                throw new InvalidOperationException($"The value {di} is out of range.");
+            Types.Buffer dst = engine.Pop<Types.Buffer>();
+            if (checked(di + count) > dst.Size)
+                throw new InvalidOperationException($"The value {count} is out of range.");
+            src.Slice(si, count).CopyTo(dst.InnerBuffer.Span[di..]);
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public virtual void Cat(ExecutionEngine engine, Instruction instruction)
+        {
+            var x2 = engine.Pop().GetSpan();
+            var x1 = engine.Pop().GetSpan();
+            int length = x1.Length + x2.Length;
+            engine.Limits.AssertMaxItemSize(length);
+            Types.Buffer result = new(length, false);
+            x1.CopyTo(result.InnerBuffer.Span);
+            x2.CopyTo(result.InnerBuffer.Span[x1.Length..]);
+            engine.Push(result);
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public virtual void SubStr(ExecutionEngine engine, Instruction instruction)
+        {
+            int count = (int)engine.Pop().GetInteger();
```

Comment:
This will change the current behavior

Reply:
The **ONLY** thing that happens is if an `overflow` or `underflow` happens it will throw an `OverflowException` for `overflow` or `InvalidOperationException` for `underflow`.

Reply:
> Also if you dont care about this. 

Don't confuse that I don't care with this being the place to solve it, open another PR, the logic is exactly the same as previously, this pr is about versioning the opcodes, your issue can be discussed in a different issue, it is not related to the jump table.

Anyways:

```csharp
BigInteger amount = BigInteger.Parse(ulong.MaxValue.ToString()); // i64
		
//Console.WriteLine("Parse: " + checked(((int)amount))); // i32 - checked
Console.WriteLine("Parse: " + ((int)amount)); // i32
```

Both lines throw `OverflowException`.

Reply:
I don't mean you personally. I mean if it doesn't matter if it overflows than do `A` else do `B`. 

default settings in most `IDE`/`Test Frameworks` do `checked`, but to ensure that it is on since we all maybe using different one with different settings.  Once compiled `Release` it acts different.

![image](https://github.com/neo-project/neo/assets/8141309/3c22f89c-021a-402b-a274-2e107cda37fe)

Reply:
We can change it a different PR, I'm not against this change

Reply:
Changes to the current logic better be in separate PRs.